aggsender/db/aggsender_db_storage.go

```diff
@@ -133,17 +154,28 @@ func (a *AggSenderSQLStorage) GetCertificateHeadersByStatus(
 		}
 
 		// Build the WHERE clause with the joined placeholders
-		query += " WHERE status IN (" + strings.Join(placeholders, ", ") + ")"
+		condition += "status IN (" + strings.Join(placeholders, ", ") + ")"
 	}
 
 	// Add ordering by creation date (oldest first)
-	query += " ORDER BY height ASC"
+	condition += " ORDER BY height ASC"
 
+	return a.getAllCerts(nil, tableCertificate, condition, args)
+}
+
+func (a *AggSenderSQLStorage) getAllCerts(tx dbtypes.Querier, table tableName,
```

Comment:
`getAllCerts` implies you want to get all the certs from `db`. This makes more sense to me.

```suggestion
func (a *AggSenderSQLStorage) getCerts(tx dbtypes.Querier, table tableName,
```

Reply:
Also AFAIK, the `tx` param is always nil, so maybe we can simplify the function signature, by removing it?
```suggestion
func (a *AggSenderSQLStorage) getAllCerts(table tableName,
```

Reply:
I prefer to keep it, because IMO, all functions that execute a SQL must receive a `tx` to keep a coherence with the multiples queries